### ISSUE TITLE
PEOPLE: Berechnung Mitgliederjahre

### ISSUE THREAD

**Issue by mtnstar** [hitobito/hitobito_sac_cas#85]
Auf einer Person gibt es neu ein computed Attribute Mitgliederjahre (membership_years)
Die Anzahl Mitgliederjahre wird anhand der mitgliedschaftsrelevanten aktiven und inaktiven Rollen in Gruppen vom Typ Sektionsmitglieder berechnet. 

# Tech-Spec
- https://github.com/hitobito/hitobito_sac_cas/blob/master/app/models/group/sektions_mitglieder.rb
- Neue collection auf Group::SektionsMitglieder : membership_roles = [Einzel, Jugend, Familie, FreiKind, Ehrenmitglied, Beguenstigt]

# ToDo

**Comment by mtnstar** [comment]
@carlobeltrame wie im Chat schon beschrieben, könnten wir nicht auf der Rolle (Mitglied) ein Feld membership_years erstellen und dann für alle aktiven und vergangenen Rollen diesen Wert kumulieren?
Für Korrekturen gibt es dann noch ein offset Feld auf der Person selber

**Comment by carlobeltrame** [comment]
Ja, wenn wir z.B. zwei Rollentypen einführen (Sektion::Mitglied und Sektion::MitgliedZusatzsektion) und nur auf der Hauptsektions-Mitglied-Rolle dieses Feld führen, dann wird der Query deutlich einfacher verständlich und skaliert vermutlich besser. Trotzdem bleibt es ein berechnetes Feld, also die Komplexität für die Umsetzung der Personenfilter und Sortierung bleibt. Sollte aber machbar sein.
Nachteil ist, dass dann trotzdem Inkonsistenzen auftreten können wenn wir Bugs haben, also zum Beispiel überlappende Hauptsektions-Mitglieds-Rollen. Die daraus folgende falsche Mitgliederjahre-Zahl ist aber immer noch einfacher debugbar und erklärbar als bei Variante 1. Das wäre dann also eine Art Kompromisslösung.
Migrationspfad zu Variante 1 und zu Variante 2 wäre glaube ich auch gegeben.
